Git intro (#338)
Code-Your-Future/syllabus: introduction/week-2/git-workshop-transcript.md

And create some new files for my CV and cover letter:

<img alt="CV.txt file selected" src="../assets/git/2.png" width="200">

You add some starting text to both files:

<img alt="CV.txt file selected" src="../assets/git/initial-changes.gif" width="400">

This is your blank canvas. You are ready to start writing!

You make a lot of changes to both files. You now have a full cover letter, a full CV, with experience, education etc. But you realise:

- you have lost your starting "template", that had a blank CV and cover letter. If you wanted to apply for a new job you would have to re-create the folder again and prepare both files again
- if you make some changes to your CV, save them, turn off your computer, and the next day you realise you don't need them anymore, it will be really hard to undo them. You need to remember exactly what you changed, and only remove those changes.

What we really want is to save occasional "[snapshots](https://www.merriam-webster.com/dictionary/snapshot)". As we're working through our job application, we want to take a "photo" of our folder each time we make an important change, so we can go back to it if we need to.

> So let's say our next task is to add a "Skills" section to our CV. We'll start by adding some skills, maybe decide to remove some, then add some more skills again, but once we're done and satisfied that we completed the task, we'll [commit](https://www.lexico.com/en/definition/commit) to being definitely done (which means we can "take a snapshot" of our folder).

Reviewer: Minor nitpick: Would it be worth using the same online dictionary, e.g. changing this one (lexico) to merriam-webster for consistency?

> First word of the day:
> **commit**

So let's do that (go [here](https://youtu.be/YSazjz1IpOc) to play the video from the beginning):

<img alt="creating git folders" src="../assets/git/creating-folders.gif" width="400">

As you can see, each time we commit to finishing a task, we create a new folder, give it the name of our task (so we can find it easier), and copy all our files in it.

We now have these folders:

<img alt="CV.txt file selected" src="../assets/git/3.png" width="250">

> The selected one, `job-application`, is our **latest** one. The one that has all the latest changes and is the most up to date. The other ones, contain different [milestones](https://www.merriam-webster.com/dictionary/milestone?src=search-dict-box) of our work, and they're named accordingly! There's a folder for our initial template (right before we first started writing) another one for when we added "Education" to our CV etc.

Reviewer: ```suggestion
The selected one, `job-application`, is our **latest** one. It is the one that has all the latest changes and is the most up to date. The other ones, contain different [milestones](https://www.merriam-webster.com/dictionary/milestone) of our work, and they're named accordingly! There's a folder for our initial template (right before we first started writing) another one for when we added "Education" to our CV etc.
```

---

But each time we finished a task, we've been copying ALL of the files to the new folder. What if we're not ready to copy all files? What if we've added a "References" section in our CV, and it looks good, and we also changed the cover letter a bit - but we're not quite ready to commit the cover letter.

Well, we can **add** only the files we want to our commit. We'll copy just the CV to the new folder, and instead of copying the latest version of the cover letter, we'll just get it from the previous folder we committed.

> Our second important word:
> **add**

---

This is all really useful! But a bit annoying to do each time! And we're going to make a lot more changes - which means A LOT more folders. It will get confusing really fast.

Would it not be nice if we had a robot we could ask to do all this work for us? We would just tell it anytime we're ready to commit, and it would create those folders for us, somewhere hidden, where we can't see them! Something like Alexa, or Siri - _"hey Alexa, add these files, I'm ready to commit them!"_

> Turns out we do, but our robot is called Git. And whenever we want to use Git in a folder, to track our changes, we need to tell it to initialize in that folder.

Reviewer: ```suggestion
Turns out we do, but our robot is called Git. And whenever we want to use Git in a folder, to track our changes, we need to first tell it to initialize in that folder.
```

> So our third important word is:
> **init**

To put it all together, the way we'd talk to Git is:

```
git init
git add CV.txt
git commit -m "add skills"
```

Now it's your turn:

> 1. Create a new folder with `mkdir`

Reviewer: ```suggestion
1. Create a new folder with [`mkdir`](https://manpages.ubuntu.com/manpages/bionic/en/man1/mkdir.1posix.html)
```
Would it be useful to link to documentation? Maybe not if this Git section comes after the introduction to the terminal + commands
